### PR TITLE
entfga - ent extension for creating FGA relationship tuples

### PR DESCRIPTION
**Pull request by golanglemonade**
Adds the inital entfga extension to create hooks for enabling the creation, update, and deletion of relationship tuples in FGA. 

The generate functions match what was previously manually created in https://github.com/datumforge/datum/blob/main/internal/ent/hooks/orgmembers.go and can be used on any user -> object relation.

Once finalized for use in the `datum` repo, an example schema will be added here to help with tests.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,3 +10,79 @@ Go libraries to interact with [OpenFGA](https://openfga.dev/)
...
+The `entc.go` file should contain the following code:
+
+```go
+//go:build ignore
+
+package main
+
+import (
+	"log"
+	"github.com/datumforge/entfga"
```
Comment (matoszz) [inline README.md:35]: is this right? or should it be `fgax`?

File: entfga/hooks.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
+package entfga
+
+import (
+	"context"
+	"fmt"
+
+	"entgo.io/ent"
+)
+
+type Mutation interface {
```
Comment (matoszz) [inline entfga/hooks.go:10]: could you put a few comments in here on consumption / use? I'm having a hard time following the multiple interfaces on Mutation + Mutator; i'm sure this is structured this way for a reason it's just hard to tell what that is (but i'm also not that great at programming :) )

File: go.mod [hunk 3]
```diff
@@ -1,34 +1,50 @@
 module github.com/datumforge/fgax
 
-go 1.21.3
+go 1.21.5
```
Comment (matoszz) [inline go.mod:3]: any reason not to just make the jump?

**Review by matoszz — APPROVED** [review]

File: go.mod (hunk 3 above)
Comment (golanglemonade) [reply go.mod:3]: nope, will bump to 1.21.6

File: README.md (hunk 1 above)
Comment (golanglemonade) [reply README.md:35]: nope, it should be `github.com/datumforge/fgax/entfga`

**Review by matoszz — APPROVED** [review]

**Pull request merged**